pkg/cmd/cli/cmd/login.go

```diff
@@ -0,0 +1,250 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	kclientcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/cli/config"
+	"github.com/openshift/origin/pkg/cmd/util"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+)
+
+type loginOptions struct {
+	username string
+	password string
+	server   string
+	project  string
+	context  string
+}
+
+func NewCmdLogin(f *osclientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &loginOptions{}
+
+	cmds := &cobra.Command{
+		Use:   "login [--username=<username>] [--password=<password>] [--server=<server>]",
+		Short: "Logs in and returns a session token",
+		Long: `Logs in to the OpenShift server and prints out a session token.
+
+Username and password can be provided through flags, the command will
+prompt for user input if not provided.
+`,
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
Can you refactor the guts of this function out into something more like https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/config/view.go#L50?  Having a series of method calls on config objects (or some such) makes it much easier to write unit tests.